### ISSUE TITLE
[Feature Request] Emoji Compact

### ISSUE THREAD

**Issue by ArmeF97** [signalapp/Signal-Android#7523]
Add Emoji Compact

See here https://twitter.com/Emojipedia/status/973529816904413185
And here https://developer.android.com/guide/topics/ui/look-and-feel/emoji-compat.html

**Comment by RiseT** [comment]
Compact or Compat?

**Comment by ArmeF97** [comment]
Compat

**Comment by RiseT** [comment]
Just refering to the typo in the title 😃

**Comment by johanw666** [comment]
It would break Jelly Bean compatibility because this library runs only on API 19+, or another api-dependent code block and another excuse to drop support for older Android versions.

**Comment by ArmeF97** [comment]
Nope, in API lower than 19, it will act like a simple Textview, and it's a supplementary thing, just in case you didn't add the emoji yet, so I don't see the problem

**Comment by FeuRenard** [comment]
I don't find any information on how to use this with the Apple iOS emoji set. And it's Signal's goal to stay consistent between platforms afaik.

**Comment by ArmeF97** [comment]
@FeuRenard  what do you mean "need its own"? How is it working now Signal on iOS?
The thing I'm requesting it's a supplementary thing for Android users, it won't change anything for iOS users. It helps to display missing character on android, only this

**Comment by FeuRenard** [comment]
Generally there exist different emoji sets. They define the small images which are displayed on your phone when you get a message with an emoji. The sender doesn't send those little images to you, but instead something like `U+1F923` and your device recognizes: "That's an emoji! To be more precise: it's the ROFL emoji.". Your App then replaces the Unicode with the according small image out of the emoji set. [Here](https://emojipedia.org/rolling-on-the-floor-laughing/) you can see different interpretations for the ROFL emoji in different emoji sets.

One Unicode emoji can look quite different in the different sets. Therefore it makes sense to provide a consistent emoji set for the users of an app to prevent misunderstandings. Signal decided to use the Apple iOS emoji set because it's difficult to use alternative emojis in an iOS App like Signal-iOS. This means even Signal-Android currently uses Apple iOS emojis and provides a consistent emoji experience.

This also means - as Signal-Android delivers its emojis bundled with the app - there are no emoji assets "missing" with old Android versions. So: which problem should the emoji compat library solve for Signal-Android?

**Comment by johanw666** [comment]
> So: which problem should the emoji compat library solve for Signal-Android?
Maybe not supported emojies sent by sms/mms for those people who use Signal as their standard sms app?

**Comment by ArmeF97** [comment]
@FeuRenard , when a new version of Unicode is standard, people can send you new characters and you don't have them in your iOS library, simply because Apple has not created them yet. Google does is on the same day, so you can use your iOS emoji, but using Google's (for android) for missing ones. 
The missing ones are the new ones :) 
Like https://emojipedia.org/face-with-pleading-eyes/

**Comment by 2-4601** [comment]
This feature request is being discussed and tracked [here](https://community.signalusers.org/t/add-emoji-compat/2313).
